src/utils/AutoId.tsx

```diff
@@ -0,0 +1,51 @@
+/** Based heavily on https://github.com/reach/reach-ui/tree/develop/packages/auto-id */
+import React, { useState, useEffect } from 'react';
+
+let serverHandoffComplete = false;
+let theId = 0;
+const genId = () => {
+  theId += 1;
+  return theId;
+};
+
+function canUseDOM() {
+  return !!(
+    typeof window !== 'undefined' &&
+    window.document &&
+    window.document.createElement
+  );
+}
+
+const useIsomorphicLayoutEffect = canUseDOM()
```

Comment:
I'd use more descriptive name like enhancedEffect (this name is from Material UI). More importantly, I think this should be moved outside AutoId to maybe a separate util or to common TS utils. This is used already at least in Popover/Portal implementation and is not only related to Id's. Should also be updated to Popover.

Reply:
This is now changed to `useEnhancedEffect` and is located in `suomifi-ui-components/src/utils/common.ts`. 
We can name the util better if needed.